times_miro.gms

```diff
@@ -361,191 +362,254 @@ os.environ['CUBEINPUTDOM'] = 'siName,typ,dd,' + ','.join(list(gams.get('xidom'))
+singleton set gmsObj(*)      'Choice of objective function formulations'      / /; // MOD, ALT, AUTO, LIN, MOD, STD
 $onMulti
-singleton set gmsddlocation(*)                'Location of DD files'          / '' 'TIMES_Demo/model/'/;
 singleton set gmsrunlocation(*)               'Location of Run file'          / '' 'TIMES_Demo/model/demo12.run'/;
+singleton set gmsrunmode                      'whether to create a MIRO scenario or solve TIMES' /'' 'create'/;
 $offMulti
 $offExternalInput
-$endif.data
+*$endif.data
+
 $onExternalInput
+$onEps
 parameter     cubeInput(%sysEnv.CUBEINPUTDOM%) / /;
+$offEps
 * Skipped VDA DATAGDX VEDAVDD 
 $offExternalInput
+set           dd_PRC_DESC(*,*,*) /  /
+              dd_COM_DESC(*,*,*) /  /;
 $offEmpty
 
 $onExternalOutput
 alias (*,soName,sow,Vintage)
-parameter cubeOutput(scenario,soName,sow,COM_GRP,PRC,ALLYEAR,ALL_REG,Vintage,ALL_TS,UC_N);
+parameter cubeOutput(soName,sow,COM_GRP,PRC,ALLYEAR,ALL_REG,Vintage,ALL_TS,UC_N);
 $offExternalOutput
 
-$eval.set DDPREFIX gmsddlocation.te
-$eval.set RUNFILE gmsrunlocation.te
+$if not set RUNFILE $eval.set RUNFILE gmsrunlocation.te
+$if not set RUNMODE $eval.set RUNMODE gmsrunmode.te
+$if not set DDPREFIX $eval.set DDPREFIX gmsddlocation.te
 $setNames "%RUNFILE%" fp fn fe
-
 *if this file is run through Studio and command line parameter is not set, the data from the *.dd files specified above will
 *be translated into a GDX file that can be imported into MIRO
-$ifThenE sameas("x%gams.IDCGDXInput%","x")or(card(dd)=0)
+$ifThenE.runmodel sameas("x%gams.IDCGDXInput%","x")or(sameas("%RUNMODE%","create"))
 * 2b) read data from *.dd files specified above
+* FF: Shouldnt this be onEchoV?
 $onecho > "%gams.scrDir%mkdd.%gams.scrExt%"
 $onmulti
 $oneps
 $include "%mydd%"
 $offecho
 $onMulti
-$ifThenE.card_dd card(dd)=0
+$ifThenE.runmode sameas("%RUNMODE%","create")
 set           solveropt(*,*,*)  'Solver options'                                 / cplex.(scaind.0,  rerun.yes, iis.yes, lpmethod.4, baralg.1,
-                                                                                  barcrossalg.1, barorder.2, threads.8)
-                                                                                  gurobi.method.2/;
+                                                                                  barcrossalg.1, barorder.2, threads.8) /;
 singleton set gmsSolver(*)      'Solver for TIMES'                               / cplex /;
 singleton set gmsTIMESsrc(*)    'Location of TIME source'                        / '' '' /; // leave at '' for default
 scalar        gmsResLim         'Time limit for solve'                           / 1000 /;
 scalar        gmsBRatio         'Basis indicator'                                / 1 /;
 singleton set gmsRunOpt(*)      'Selection for local, short and long NEOS queue' / local /; // local, short, long
+
+*Clear data from MIRO that may cause duplicate errors when creating a scenario
+$onMultiR
```

Comment:
Why do we need this $onMultiR here?

Reply:
When creating a miro scenario we need to clear the domain set `dd`, which is only possible with `$onMultiR`. 

In the succeeding python code `dd` is filled again. Replacing `dd` is not possible there as far as I know (add_record())

Reply:
I see. We can't clear domains without `$onMultiR`.